src/utils/hooks/useLoadApplication.tsx

```diff
@@ -31,13 +35,21 @@ const useLoadApplication = (props: useLoadApplicationProps) => {
   useEffect(() => {
     if (data && data.applicationBySerial) {
       const application = data.applicationBySerial as Application
+      console.log(
+        'application.applicationStageHistories',
+        application.applicationStageHistories.nodes
+      )
+
+      const stage = application.applicationStageHistories.nodes[0] as ApplicationStageHistory
```

Comment:
I have added in the query here:
https://github.com/openmsupply/application-manager-web-app/pull/205/files#diff-179591f74393632ba97608c9fb201ce872cb9e23b2a9b04dfdf083155b3258d6
What I don't like it still returns as an array. That's why I was asking you if we could add more things to the computed column `stage` with what I will need here which is only the `stage_id`.
And I think I still need to add a checking just for unexpected length === 0 here.

Reply:
Yeah, but the first index of the array is not necessarily the one we want. I just tested by making a new application_stage_history for an existing application, and this query doesn't retrieve the `is_current` one. 

Suggestion: rather than make a new computed column, or dive deep into the tree like here, why not just add another "node" to the main query to query the `application_stage_status_all` View at the same time? That's what that view is intended for -- quick access to all stage and status details for a given application.

Reply:
![Screen Shot 2020-12-15 at 9 27 15 AM](https://user-images.githubusercontent.com/5456533/102132085-055fc880-3eb8-11eb-93d4-04070bf6d297.png)
Like this. Although I'm fine with adding additional computed columns to application table if you think that's easier?

Reply:
I agree. That's what I was looking for, but haven't thought. Thanks for this advise! I'll change to use that now.